### ISSUE TITLE
Analytics CORS issue

### ISSUE THREAD

**Issue by aquelehugo** [BuilderIO/partytown#43]
I did my implementation exactly like your example, but instead of using Google Tag Manager, I am using the Google Analytics snippet:

```
    <script type="text/partytown">
      (function (i, s, o, g, r, a, m) {
        i['GoogleAnalyticsObject'] = r;
        (i[r] =
          i[r] ||
          function () {
            (i[r].q = i[r].q || []).push(arguments);
          }),
          (i[r].l = 1 * new Date());
        (a = s.createElement(o)), (m = s.getElementsByTagName(o)[0]);
        a.async = 1;
        a.src = g;
        m.parentNode.insertBefore(a, m);
      })(
        window,
        document,
        'script',
        'https://www.google-analytics.com/analytics.js',
        'ga'
      );
    </script>
    <script type="text/partytown" src="https://securepubads.g.doubleclick.net/tag/js/gpt.js"></script>
    <script>
      partytown = {
        forward: ['ga']
      };
    </script>
    <script>/* ---- Partytown 0.0.30 inlined ---- */</script>
```

When I do that, I get

```Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://www.google-analytics.com/analytics.js. (Reason: CORS header ‘Access-Control-Allow-Origin’ missing). Status code: 200.```

Am I missing something?

**Comment by adamdbradley** [comment]
For any CORS issues, you'll need to setup a proxy the correct headers can be added by the server: https://github.com/BuilderIO/partytown/wiki/Proxying-Requests#configuring-url-proxies

**Comment by johnfmorton** [comment]
I had CORS issues and read the [link about proxies](https://github.com/BuilderIO/partytown/wiki/Proxying-Requests#configuring-url-proxies) above. I have a working solution, but I'd like feedback. 

Let's say my site is at `https://example.com`. I set up a new domain at `proxy.example.com` to act as the proxy for the primary site's Partytown calls. I've got a single file, `index.php`, in the webserver at `proxy.example.com`. Here is the content of that file:

https://github.com/BuilderIO/partytown/wiki/Proxying-Requests#configuring-url-proxies

Also, in the nginx.config for the `proxy.example.com` server, I have added a header to allow `example.com`.

```
add_header 'Access-Control-Allow-Origin' 'https://example.com';
```

In my tests, this proxy does the job. I'm seeing the analytics coming back from Google Analytics that I expect. 

But what am I missing? Here's are the guardrails I've got in place.

* I'm checking the headers of incoming requests and only allowing the proxy to be used when I see what I expect.
* I only have "allow origin" headers for the server I want to use the proxy. 
* I'm logging metrics (in a very basic way) to a file to see what's happening to the proxy.

**How could this proxy be abused? How can it be improved?**

Any feedback is greatly appreciated. 

Thanks.
